plutus-core/plutus-ir/src/PlutusIR/Transform/LetFloatIn.hs

```diff
@@ -125,12 +141,16 @@ whose LHS is used more than once, and we descend in all other cases;
 -}
 
 -- | The uniques of all used variables in a term.
-type Uniques = Set PLC.TermUnique
+type Uniques = (Set PLC.TermUnique, Set PLC.TypeUnique)
```

Comment:
I'm not sure we really need to track these separately. Renaming ensures _global_ uniqueness across term and type uniques, so it's safe to just throw them into a single set of `Unique`s. That would simplify your life quite a bit.

Reply:
Right, I just realized there's a `PLC.theUnique`